### ISSUE TITLE
Wrong differentiation result for oneflow.nn.FakeQuantization

### ISSUE THREAD

**Issue by xxxyyyzzz12345** [Oneflow-Inc/oneflow#8649]
## Summary

The result of the gradient computed by the following code snippet is always 1.0.

However, according to the definition of oneflow.nn.FakeQuantization in official documentation, shouldn't the gradient be 0 instead of 1?

## Code to reproduce bug
```
quantization_formula ='google'
quantization_bit = 8
quantization_scheme = 'symmetric'
input1 = oneflow.rand(1, dtype=oneflow.float64,requires_grad=True)
input2 = oneflow.tensor([1], dtype=oneflow.float64,requires_grad=True)
input3 = oneflow.tensor([0], dtype=oneflow.float64,requires_grad=True)
mod = oneflow.nn.FakeQuantization(quantization_formula, quantization_bit, quantization_scheme)
output = mod(input1, input2, input3)
output.backward()
print(input1.grad)
```
## System Information

- What is your OneFlow installation (pip, source, dockerhub): pip
- OS: Ubuntu 20.04.2 LTS
- OneFlow version (run `python3 -m oneflow --doctor`): 0.7.0+cu112
- Python version: 3.8.8
- CUDA driver version: 11.4·

**Comment by BBuf** [comment]
> ## Summary
> 
> The result of the gradient computed by the following code snippet is always 1.0.
> 
> However, according to the definition of oneflow.nn.FakeQuantization in official documentation, shouldn't the gradient be 0 instead of 1?
> ## Code to reproduce bug
> 
> ```
> quantization_formula ='google'
> quantization_bit = 8
> quantization_scheme = 'symmetric'
> input1 = oneflow.rand(1, dtype=oneflow.float64,requires_grad=True)
> input2 = oneflow.tensor([1], dtype=oneflow.float64,requires_grad=True)
> input3 = oneflow.tensor([0], dtype=oneflow.float64,requires_grad=True)
> mod = oneflow.nn.FakeQuantization(quantization_formula, quantization_bit, quantization_scheme)
> output = mod(input1, input2, input3)
> output.backward()
> print(input1.grad)
> ```
> 
> ## System Information
> 
>     * What is your OneFlow installation (pip, source, dockerhub): pip
> 
>     * OS: Ubuntu 20.04.2 LTS
> 
>     * OneFlow version (run `python3 -m oneflow --doctor`): 0.7.0+cu112
> 
>     * Python version: 3.8.8
> 
>     * CUDA driver version: 11.4·

The gradient of FakeQuantization is always 1.0, beacase we use [STRAIGHT-THROUGHESTIMATOR](https://arxiv.org/pdf/1606.06160.pdf) . If the  gradient is 0.0, the network parameters will never be updated which is obviously unreasonable.

**Comment by xxxyyyzzz12345** [comment]
But when the input is outside the range [quant_min,quant_max], shouldn't the gradient be 0.0 instead of 1.0?
The following code snippet sets both quant_min and quant_max to 0 and defines the input tensor as a 1-d tensor ranging from -20 to 20 with a step size of 1:
```
quantization_formula ='google'
quantization_bit = 1
quantization_scheme = 'symmetric'

for x in range(-20,20):
    input1 = oneflow.tensor([x], dtype=oneflow.float64,requires_grad=True)
    input2 = oneflow.tensor([1], dtype=oneflow.float64,requires_grad=True)
    input3 = oneflow.tensor([0], dtype=oneflow.float64,requires_grad=True)
    mod = oneflow.nn.FakeQuantization(quantization_formula, quantization_bit, quantization_scheme)
    output = mod(input1, input2, input3)
    output.backward()
    print(output)
    print(input1.grad)
```

Both the outputs (-1, ..., -1, 0, ..., 0) and gradients (1., 1., 1., ...) seem to be incorrect.
